app/models/petition.rb

```diff
@@ -156,6 +155,13 @@ def self.counts_by_state
     counts_by_state
   end
 
+  def publish!
+    self.state = Petition::OPEN_STATE
+    self.open_at = Time.zone.now
+    self.closed_at = AppConfig.petition_duration.months.from_now.end_of_day
```

Comment:
It is ``<``, but on relfection I think I'll leave it as-is.  If we make it the ``+ 1.day).midnight`` version then we'll have to do weird things when displaying the closing date.  It might be correct that midnight is the start of the day, not the end of the day, but I think if you said to someone that the petition closed at midnight on a given day, they'd think the end, not the start.  So it's easier to have the ``.end_of_day`` version stored, than the ``.midnight`` version in terms of display logic.